flow/cmd/cert.go

```diff
@@ -26,3 +32,91 @@ func parseTemporalCertAndKey(ctx context.Context) ([]tls.Certificate, error) {
 
 	return []tls.Certificate{keyPair}, nil
 }
+
+func setupTemporalClient(ctx context.Context, clientOptions client.Options) (client.Client, error) {
+	if certPath := internal.PeerDBTemporalClientCertPath(); certPath != "" {
+		slog.Info("Using temporal certificate/key from paths for authentication")
+		keyPath := internal.PeerDBTemporalClientKeyPath()
+		var certMtime, keyMtime time.Time
+		var keyPair *tls.Certificate
+
+		clientOptions.ConnectionOptions = client.ConnectionOptions{
+			TLS: &tls.Config{
+				GetClientCertificate: func(*tls.CertificateRequestInfo) (*tls.Certificate, error) {
+					if keyPair != nil {
+						if certInfo, err := os.Stat(certPath); err == nil && certInfo.ModTime().Equal(certMtime) {
+							if keyInfo, err := os.Stat(keyPath); err == nil && keyInfo.ModTime().Equal(keyMtime) {
+								return keyPair, nil
+							}
+						}
```

Comment:
Secrets are mounted as 1/2 level symlinks and os.Stat might not work

Anyways Temporal uses a grpc connection (so I assume single TCP connection) and secrets are mounted in a ramdisk in kubernetes as far as I know so I assume it should not be an issue to simply re-read